src/services/tokens.service.ts

```
@@ -0,0 +1,74 @@
+import { strict as assert } from 'assert';
+import { web3Factory } from '@arianee/spkz-sdk';
+import sharedTokenAbi from '../abi/shared-token-abi';
+
+interface TokenInfo {
+  success: boolean;
+  details?: string;
+  name?: string;
+  symbol?: string;
+  decimals?: number;
+}
+
+/**
+ * Returns simple token infos if the address
+ * is an ERC-20 contract or an ERC-721 contract, throws otherwise
+ * @param parameters chainId and address of the contract
+ * @returns name, symbol and decimals of the token
+ */
+export const tryGetTokenInfos = async (parameters: {
+  chainId: string;
+  address: string;
+}): Promise<TokenInfo> => {
+  const { chainId, address } = parameters;
+
+  assert(chainId, 'chainId must be defined');
+  assert(address, 'contract address must be defined');
+
+  let web3Provider;
+  try {
```

If web3provider throws and we simply let it throws there, the route will return a HTTP 400 (chain not found), although it should be a 404, this is why we return an object with success: false here.

There is no guarantee that web3provider won't throw other errors in the future and since it uses a generic Error object, we can't filter the catch based on the error type, question : do we keep it that way to have a correct 404 or do we simply do
`const web3Provider = web3Provider = await web3Factory(chainId);` 
and let it throw, resulting in a HTTP 400 whatever the reason is?